packages/plugins/response-cache/src/plugin.ts

```diff
@@ -410,7 +411,10 @@ export function useResponseCache<PluginContext extends Record<string, any> = {}>
 
           types.add(typename);
           if (typename in ttlPerType) {
-            currentTtl = calculateTtl(ttlPerType[typename], currentTtl);
+            const maybeTtl = ttlPerType[typename] as number | undefined;
+            if (maybeTtl != null && Number.isNaN(maybeTtl) === false) {
+              currentTtl = calculateTtl(maybeTtl, currentTtl);
+            }
```

Comment:
To avoid duplication, perhaps`calculateTtl` can do this check by taking either `unknow` or `number | undefined | null` as a first parameter ?

Reply:
Fixed, pushed a new commit. Please mark as resolved if it looks good.